### PR TITLE
#160577482 Implement comments(CRUD)

### PR DESCRIPTION
**Pull request by gloriaodipo**
**What does this PR do?**
- This PR implements CRUD functionality for comments feature

**Description of task to be completed**
-Only authenticated users should be able to comment on articles.
- Successive comments should be threaded
- Comments should be timestamped
- Any user can get comments
- Only comment owners should be able to delete and update comments

**Endpoints**
`articles/<slug>/comments/`
`articles/<slug>/comments/<int:id>/`

### CODE REVIEW

File: authors/apps/articles/admin.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 from django.contrib import admin
-from .models import ArticlesModel
+
+from .models import ArticlesModel,Comment
```
Comment (FawazFarid) [inline authors/apps/articles/admin.py]: Insert a space between the comma and Comment

File: authors/apps/articles/models.py [hunk 2]
```diff
@@ -37,3 +37,28 @@ def save(self, *args, **kwargs):
...
+   created_at = models.DateTimeField(auto_now_add=True)
+   updated_at = models.DateTimeField(auto_now=True)
+   author = models.ForeignKey(User, related_name='comments', on_delete=models.CASCADE)
+   article = models.ForeignKey(ArticlesModel, related_name='comments', on_delete=models.CASCADE)
+   parent = models.ForeignKey('self',
+               null=True,
+               blank=True,
+               related_name='threads',
+               on_delete=models.CASCADE
+               )
```
Comment (FawazFarid) [inline authors/apps/articles/models.py]: the indentation for this seems a bit off to me, I'd do it like this :
```
parent = models.ForeignKey(
   'self',
    null=True,
    blank=True,
    related_name='threads',
    on_delete=models.CASCADE
)
```
I'm not super sure but this looks more readable 😅, what do you think?

File: authors/apps/articles/serializers.py [hunk 3]
```diff
@@ -52,4 +52,42 @@ class Meta:
...
+       """
+       threads = [
+           {
 
+               'id': thread.id,
+                'body': thread.body,
+                'author': thread.author.username,
+                'created_at': self.format_date(thread.created_at),
+                'replies': thread.threads.count(),
+                'updated_at': self.format_date(thread.updated_at)}  for thread in instance.threads.all()]
```
Comment (FawazFarid) [inline authors/apps/articles/serializers.py]: the closing curly bracker `}` should be in the next line, and the `]` should be on the next line after e.g
```
...
        'updated_at': self.format_date(thread.updated_at)
    } for thread in instance.threads.all()
]

```

**Review by FawazFarid — CHANGES_REQUESTED** [review]

File: authors/apps/articles/admin.py (hunk 1 above)
Comment (gloriaodipo) [reply authors/apps/articles/admin.py]: Noted

File: authors/apps/articles/models.py (hunk 2 above)
Comment (gloriaodipo) [reply authors/apps/articles/models.py]: Yeah, sure 😅

**Review by FawazFarid — APPROVED** [review]

**Pull request merged**